Plugins/NativeCamera/Source/NativeCameraAndroid.cpp

```diff
@@ -0,0 +1,404 @@
+        {
+            const char *id = cameraIds->cameraIds[i];
+
+            ACameraMetadata *metadataObj;
+            ACameraManager_getCameraCharacteristics(cameraManager, id, &metadataObj);
+
+            ACameraMetadata_const_entry lensInfo = {};
+            ACameraMetadata_getConstEntry(metadataObj, ACAMERA_LENS_FACING, &lensInfo);
+
+            auto facing = static_cast<acamera_metadata_enum_android_lens_facing_t>(lensInfo.data.u8[0]);
+
+            // Found a corresponding facing camera?
+            if (facing == (frontCamera ? ACAMERA_LENS_FACING_FRONT : ACAMERA_LENS_FACING_BACK))
+            {
+                cameraId = id;
+                break;
+            }
+        }
+
+        ACameraManager_deleteCameraIdList(cameraIds);
+        return cameraId;
+    }
+
+    // device callbacks
+    static void onDisconnected(void* /*context*/, ACameraDevice* /*device*/)
+    {
+    }
+
+    static void onError(void* /*context*/, ACameraDevice* /*device*/, int /*error*/)
+    {
+    }
+
+    static ACameraDevice_stateCallbacks cameraDeviceCallbacks = {
+            .context = nullptr,
+            .onDisconnected = onDisconnected,
+            .onError = onError
+    };
+
+    // session callbacks
+    static void onSessionActive(void* /*context*/, ACameraCaptureSession* /*session*/)
+    {
+    }
+
+    static void onSessionReady(void* /*context*/, ACameraCaptureSession* /*session*/)
+    {
+    }
+
+    static void onSessionClosed(void* /*context*/, ACameraCaptureSession* /*session*/)
+    {
+    }
+
+    static ACameraCaptureSession_stateCallbacks sessionStateCallbacks {
+        .context = nullptr,
+        .onClosed = onSessionClosed,
+        .onReady = onSessionReady,
+        .onActive = onSessionActive
+    };
+
+    // capture callbacks
+    static void onCaptureFailed(void* /*context*/, ACameraCaptureSession* /*session*/, ACaptureRequest* /*request*/, ACameraCaptureFailure* /*failure*/)
+    {
+    }
+
+    static void onCaptureSequenceCompleted(void* /*context*/, ACameraCaptureSession* /*session*/, int /*sequenceId*/, int64_t /*frameNumber*/)
+    {
+    }
+
+    static void onCaptureSequenceAborted(void* /*context*/, ACameraCaptureSession* /*session*/, int /*sequenceId*/)
+    {
+    }
+
+    static void onCaptureCompleted (void* /*context*/, ACameraCaptureSession* /*session*/, ACaptureRequest* /*request*/, const ACameraMetadata* /*result*/)
+    {
+    }
+
+    static ACameraCaptureSession_captureCallbacks captureCallbacks {
+        .context = nullptr,
+        .onCaptureStarted = nullptr,
+        .onCaptureProgressed = nullptr,
+        .onCaptureCompleted = onCaptureCompleted,
+        .onCaptureFailed = onCaptureFailed,
+        .onCaptureSequenceCompleted = onCaptureSequenceCompleted,
+        .onCaptureSequenceAborted = onCaptureSequenceAborted,
+        .onCaptureBufferLost = nullptr,
+    };
+
+    arcana::task<void, std::exception_ptr> CheckCameraPermissionAsync()
```

Comment:
Should we factor this out of xr.cpp like you did some of the OpenGL related stuff?